### DISCUSSION TITLE
Problemi integrazione haaska e AWS su HA (Docker)

### DISCUSSION THREAD

**Discussion by dbarbisan** [indomus/forum#1783] [HUB: Home Assistant]
Buon giorno a tutti,
seguendo con molta attenzione la guida ho provato più volte ad eseguire l'integrazione di Alexa in oggetto. Pur non essendo esperto, la procedura non mi è nuova perché già fatta in passato con successo su altre mie installazioni . Ora sono passato ad una nuova installazione di HA su docker (la migrazione non mi era riuscita!) e quindi ho configurato Caddy, che mi pare funzionare correttamente (in particolare l'URL esterno). 
L'errore che mi viene restituito in fase di test è questo:

```
{
  "errorMessage": "401 Client Error: Unauthorized for url: https://cas***.duckdns.org:***/api/alexa/smart_home",
  "errorType": "HTTPError",
  "requestId": "***",
  "stackTrace": [
    "  File \"/var/task/haaska.py\", line 111, in event_handler\n    return ha.post('alexa/smart_home', event, wait=True)\n",
    "  File \"/var/task/haaska.py\", line 66, in post\n    r.raise_for_status()\n",
    "  File \"/var/task/requests/models.py\", line 940, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"
  ]
}
```
Contemporaneamente sul log di HA mi compare l'errore:

```
Login attempt or request with invalid authentication from ***.eu-west-1.compute.amazonaws.com (***). (Home Assistant Alexa Smart Home Skill - eu-west-1 - python-requests/2.21.0)
```
Qualche suggerimento? Grazie...

**Comment by pierraffaele80** [comment]
Medesimo problema... nessuno sa come si risolve?

**Comment by KC-inDomus** [comment]
`"errorMessage": "401 Client Error: Unauthorized for url: https://cas***.duckdns.org:***/api/alexa/smart_home"`

è un errore abbastanza chiaro. indagate sul tema token e autenticazione

**Reply by pierraffaele80** [reply]
credimi, sono tre giorni che verifico l'impossibile, seguendo la guida di indomus...
non so dove sbattere la testa

**Reply by snakuzzo** [reply]
dai un'occhiata sulle issue sul progetto di haaska.
Guarda [QUI](https://github.com/mike-grant/haaska/issues/170) ad esempio

**Reply by pierraffaele80** [reply]
Grazie, corro a vedere

**Comment by pierraffaele80** [comment]
Capisco che il problema è causato dal reverse proxy.
io, ad esempio non ho il file ip_bans.yaml

**Reply by snakuzzo** [reply]
che usi come reverse proxy ?
che configurazione hai ?

P.S. Ti pregherei di fare reply sotto l'ultima risposta e non creare nuovi thread (come hai fatto ora) ...e di non fare reply direttamente dalla mail

**Reply by pierraffaele80** [reply]
scusami, ho risposto, prima via mail. Questo ha creato il nuovo thread.

Utilizzo ha core su docker container (Raspbian) e caddy

**Reply by snakuzzo** [reply]
e la config di Caddy ?
Questa è la mia e non ho alcun problema con haaska...

```
(https_header) {
	header {
		Strict-Transport-Security "max-age=31536000; includeSubdomains"
		X-XSS-Protection "1; mode=block"
		X-Content-Type-Options "nosniff"
		X-Frame-Options "SAMEORIGIN"
		Referrer-Policy "same-origin"
	}
} 

{$HOMEASSISTANT_EXTERNAL_URL} {
	import https_header
	reverse_proxy {$HOMEASSISTANT_LOCAL_URL}
}
```

**Reply by pierraffaele80** [reply]
```
header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}

https://******************.duckdns.org {
  respond "Saluti dal Pier!"
  file_server
}
https://******************.duckdns.org:9000 {
  import https_header
  reverse_proxy http://192.168.178.4:8123
}
```

**Reply by pierraffaele80** [reply]
ho omesso la prima parte del mio indirizzo .duckdns.org

**Reply by snakuzzo** [reply]
per favore correggi usando il markdown, non si capisce una fava :D

Come usare il markdown -> https://github.com/indomus/forum/discussions/1311

**Reply by snakuzzo** [reply]
ad ogni modo...assicurati che il config.json della [lambda function](https://eu-west-1.console.aws.amazon.com/lambda) sia corretto (di seguito come dovrebbe essere). 

![image](https://user-images.githubusercontent.com/7205371/208628756-7b20a6bc-e611-4c13-a4a6-3ef88edafbb9.png)

Più che altro assicurati che il token sia ok, perché potrebbe anche essere quello il problema.

Poi assicurati di fare il DEPLOY dopo ogni modifica e solo dopo il TEST

**Reply by pierraffaele80** [reply]
Li ho già errore. Stavo rifacendo tutti i passaggi; ti posto l'errore tra un attimo

**Reply by snakuzzo** [reply]
In quello che hai postato di caddy manca una riga all'inizio. Confronta bene il mio con il tuo

**Reply by pierraffaele80** [reply]
Non capisco cosa sia successo, ma... ha funzionato.

Si è collegato nuovamente ad HA! Forse era il token, come dici tu... La stringa sul caddyfile c'era, mancava nella porzione di codice che ho incollato.

Più che altro mi spiace non aver capito dove commettevo l'errore..

Grazie per il supporto ed il sostegno!

**Reply by snakuzzo** [reply]
figurati ;)

**Reply by snakuzzo** [reply]
PIù che altro, per la prossima volta, quello evidenziato nel log di Core è il codice [HTTP 401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401). Se segui il link, potrai constatare anche tu che doveva essere per forza di cose essere un problema legato alle credenziali.
